File: 0392-is-subsequence/0392-is-subsequence.java
class Solution {
    public boolean isSubsequence(String str, String ttr) {
        char[] s = str.toCharArray();
        char[] t = ttr.toCharArray();
        int n = s.length ,m = t.length;
        int i = 0 , j = 0;
        while(i < n && j < m){
            if(s[i] == t[j]){
                i += 1;
                j += 1;
            }else
                j += 1;
        }

        return i >= n;
    }
}

File: 0392-is-subsequence/0392-is-subsequence.cpp
class Solution {
public:
    int m,n;
    int dp[100032][132];
    bool rec(int i,int j,string t,string s){

        if(dp[i][j] != -1)
            return dp[i][j];
        if(n == j){
            return true;
        }
        if(m == i)
            return false;
        
        if(t[i] == s[j]){
            if(rec(i+1,j+1,t,s))
                return dp[i][j] = true;
        }else if(rec(i+1,j,t,s))
            return dp[i][j] = true;

        return dp[i][j] = false;

    }

    bool isSubsequence(string s, string t) {
        m = t.size();
        n = s.size();
        memset(dp,-1,sizeof(dp));
        return rec(0,0,t,s);    
    }
};